test/built-ins/String/prototype/search/cstm-search-invocation.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invocation of @@search property of user-supplied objects
+es6id: 21.1.3.15
+info: >
+    [...]
+    3. If regexp is neither undefined nor null, then
+       a. Let searcher be GetMethod(regexp, @@search).
+       b. ReturnIfAbrupt(searcher).
+       c. If searcher is not undefined, then
```

Comment:
@jugglinmike, could you add another test to verify that the built-in search is used if `searcher` is `undefined`?